daemon/api.go

```diff
@@ -281,6 +281,11 @@ func getSnapsInfo(c *Command, r *http.Request) Response {
 		includeLocal = true
 	}
 
+	var includeTypes []string
+	if len(query["types"]) > 0 {
```

Comment:
Sorry, I expressed this badly. Its fine of course to filter on more than one type.

The code checks for `len(query(["types"]) > 0` and then uses `query["types"][0]`. But what if there is a second "types" parameter in the query (or multiple ones)? I.e. something like:
```
+func (s *apiSuite) TestSnapsInfoMultipleQ(c *check.C) {
+       s.mkInstalled(c, "app", "foo", "v1", true, "type: app")
+       s.mkInstalled(c, "framework", "foo", "v1", true, "type: framework")
+
+       req, err := http.NewRequest("GET", "/2.0/snaps?types=framework&types=app", nil)
...
```

It seems like we have various options: ignore silently, return error, merge via something like:
```
	var includeTypes []string
	for _, q := range query["types"] {
		includeTypes = append(includeTypes, strings.Split(q, ",")...)
	}
```
I don't really know what client developers would expect usually. Maybe it is something that no one would send to the API anyway so not worth bothering :)